core/ArchiveProcessor/Loader.php

```diff
@@ -94,6 +103,16 @@ private function prepareArchiveImpl($pluginName)
             return $idArchive;
         }
 
+        // NOTE: this optimization helps when archiving large periods. eg, if archiving a year w/ a segment where
+        // there are not visits in the entire year, we don't have to go through and do anything. but, w/o this
+        // code, we will end up launching archiving for each month, week and day, even though we don't have to.
+        //
+        // we don't create an archive in this case, because the archive may be in progress in some way, so a 0
+        // visits archive can be inaccurate in the long run.
+        if ($this->canSkipThisArchive()) {
+            return false;
+        }
```

Comment:
Explanation part 8: here is the next place the canSkipThisArchive() optimization is used. Doing it here means we don't even have to launch archiving as much for large periods. For example, if there is a year w/ one visit, we just launch archiving for the year, 1 month, 1 week and 1 day at most. Rather than 1 year, 12 months, 52 weeks, 365 days.